components/brave_extension/extension/brave_extension/background/reducers/ampReducer.ts

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as webNavigationTypes from '../../constants/webNavigationTypes'
+import { Actions } from '../../types/actions/index'
+import * as ampAPI from '../api/ampAPI'
+
+export default function ampReducer (state = {}, action: Actions) {
+  switch (action.type) {
+    case webNavigationTypes.ON_COMMITTED: {
+      if (action.isMainFrame) {
+        ampAPI.getAMPNeutralized().then(ampNeutralized => {
+          if (ampNeutralized) {
+            chrome.tabs.executeScript(action.tabId, {
```

Comment:
we can do the redirect in the native code, we really want to avoid using extensions for stuff like this. Check out brave_common_static_redirect_network_delegate_helper

Reply:
so actually in moving this to greaselion we will just conditionally enable this content script when the pref values changes

Reply:
same with this https://github.com/brave/brave-core/pull/4950/files#diff-e5fa22746ae0f1e347e9e93cd7113627R9